Source/KeyHandler.mm

```diff
@@ -531,7 +533,19 @@ - (BOOL)handleInput:(KeyHandlerInput *)input state:(InputState *)inState stateCa
                 return YES;
             }
         }
+
+        size_t originalCursorIndex = _grid->cursor();
+
+        // Note: When the cursor is at the end of the composing buffer and the
+        // preference that make McBopomofo be like MS Bopomofo are on, the
+        // cursor should be moved to the begin of the last character.
+        if (originalCursorIndex == _grid->length() &&
+            Preferences.selectPhraseAfterCursorAsCandidate &&
+            Preferences.moveCursorAfterSelectingCandidate) {
+            _grid->setCursor(originalCursorIndex - 1);
```

Comment:
What would happen if `originalCursorIndex` is 0?

Reply:
Since originalCursorIndex is now as the length of the grid, and it cannot enter the candidate state if the length of the grid is zero, I do not thinkoriginalCursorIndex could be zero.

Reply:
Makes sense. Thanks.